doc/source/overview/high_level.md

```diff
@@ -0,0 +1,508 @@
+
+Among the explainers in this section, there are two approaches taken. The first fits a counterfactual when the user requests the insight. This happens during the `.explain()` method call on the explainer class. This is done by running gradient descent on model inputs to find a counterfactual. The methods that take this approach are counterfactual instances, contrastive explanation, and counterfactuals guided by prototypes methods. Thus, the `fit` method in these cases are quick, but the `explain` method is slow.
+
+The other approach, however, uses reinforcement learning to train a model that produces explanations on demand. The training takes place during the `fit` method call, so this has a long runtime while the `explain` method is quick. If you want performant explanations in production environments, then the latter approach is preferable.
+:::
+
+__TODO__:
+- schematic image explaining search for counterfactual as determined by loss
+- schematic image explaining difference between different approaches.
+
+#### Counterfactuals Instances
+
+| Model-types | Task-types     | Data-types  |
+| ----------- | -------------- | ----------- |
+| TF/Kera     | Classification | Tabular     |
+| Black-box   |                | Image       |
+
+Let the model be given by $f$, and let $p_t$ be the target probability of class $t$. Let $0<\lambda<1$ be a hyperparameter. This method constructs counterfactual instances from an instance $X$ by running gradient descent on a new instance $X'$ to minimize the following loss.
+
+$$L(X', X)= (f_{t}(X') - p_{t})^2 + \lambda L_{1}(X', X)$$
+
+The first term pushes the constructed counterfactual towards the desired class, and the use of the $L_{1}$ norm encourages sparse solutions.
+
+This method requires computing gradients of the loss in the model inputs. If we have access to the model and the gradients are available, this can be done directly. If not, we can use numerical gradients, although this comes at a considerable performance cost.
+
+A problem arises here in that encouraging sparse solutions doesn't necessarily generate interpretable counterfactuals. This happens because the loss doesn't prevent the counterfactual solution from moving off the data distribution. Thus, you will likely get an answer that doesn't look like something that you'd expect to see from the data.
+
+__TODO:__
+- Picture example. Something similar to: https://github.com/interpretml/DiCE
+
+**Pros**
+- Both a black and white box method
+- Doesn't require access to the training dataset
+
+**Cons**
+- Not likely to give human interpretable instances
+- Requires configuration in the choice of $\lambda$
+
+
+#### Contrastive Explanation Method
```

Comment:
Let's discuss if we want to name this Pertinent Negatives. Maybe best "Contrastive Explanation Method (Pertinent Negatives)".